torch/utils/cpp_extension.py

```diff
@@ -575,10 +575,12 @@ def __init__(self, *args, **kwargs):
         return cls_with_options
 
     def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
         self.no_python_abi_suffix = kwargs.get("no_python_abi_suffix", False)
-
         self.use_ninja = kwargs.get('use_ninja', True)
+
+        filtered_kwargs = {kw: val for kw, val in kwargs.items() if kw not in ["no_python_abi_suffix", "use_ninja"]}
```

Comment:
Nit should use tuple or set for contains check. Set should be optimized since it's a check against literal strings.